vm/src/slots.rs

```diff
@@ -35,12 +36,22 @@ impl Default for PyTpFlags {
 
 #[derive(Default)]
 pub struct PyClassSlots {
-    pub name: Option<String>, // tp_name, not class name
+    pub flags: PyTpFlags,
+    pub name: PyRwLock<Option<String>>, // tp_name, not class name
     pub new: Option<PyNativeFunc>,
     pub call: Option<PyNativeFunc>,
     pub descr_get: Option<PyDescrGetFunc>,
 }
 
+impl From<PyTpFlags> for PyClassSlots {
```

Comment:
not sure `From` trait fits here. it look like construct with flags rather than conversion from flags.